server/plugin_release.go

```diff
@@ -750,53 +751,57 @@ func getPluginSigningSftpClient(cfg *MatterbuildConfig) (*sftp.Client, error) {
 	return sftp, nil
 }
 
-// hasExpectedPlatformBinaries verifies if plugin tar contains at least the binaries expected
-func hasExpectedPlatformBinaries(filePath string, expectedBinaries map[string]string) error {
-	f, err := os.Open(filePath)
+// findPlatformBinaries finds the binaries for which the plugin was compiled
+func findPlatformBinaries(filePath string) (map[string]string, error) {
+	tmpDir, err := os.MkdirTemp("", "platform-plugin-*")
 	if err != nil {
-		return errors.Wrap(err, "failed to setup sftp client")
+		return nil, errors.Wrap(err, "failed to create temporary directory")
 	}
-	defer f.Close()
+	defer os.RemoveAll(tmpDir) // clean up
 
-	gzf, err := gzip.NewReader(f)
+	err = unpackPlugin(filePath, tmpDir)
 	if err != nil {
-		return errors.Wrap(err, "failed to read gzip data")
+		return nil, errors.Wrap(err, "failed to unpack plugin")
 	}
 
-	foundBinaries := make(map[string]bool, len(expectedBinaries))
-	for _, v := range expectedBinaries {
-		foundBinaries[v] = false
+	dir, err := os.ReadDir(tmpDir)
```

Comment:
This "pre-invocation setup" of `model.FindManifest` is copied right out of the server. It really should be part of `model.FindManifest` itself.